src/mds/MDSRank.cc

```diff
@@ -993,7 +993,10 @@ void MDSRank::ProgressThread::shutdown()
 
 bool MDSRankDispatcher::ms_dispatch(const cref_t<Message> &m)
 {
-  if (m->get_source().is_client()) {
+  if (m->get_source().is_mds()) {
+    ceph_assert(dynamic_cast<const MMDSOp*>(m->get()));
```

Comment:
Should this not be `m.get()`? That also avoids the new method you've added in `src/msg`.

Reply:
Fixed